src/flight.rs

```diff
@@ -75,36 +87,33 @@ impl SqlFlightClient {
                 Err(ArrowError::ParseError("Invalid auth header!".to_string()))?;
             }
             let auth = auth[bearer.len()..].to_string();
-            self.token = Some(auth);
+            token = Some(auth);
         }
-        let responses: Vec<HandshakeResponse> = resp
-            .into_inner()
-            .try_collect()
-            .await
-            .map_err(|_| ArrowError::ParseError("Can't collect responses".to_string()))?;
-        let resp = match responses.as_slice() {
-            [resp] => resp.payload.clone(),
-            [] => Bytes::new(),
-            _ => Err(ArrowError::ParseError(
-                "Multiple handshake responses".to_string(),
-            ))?,
-        };
-        Ok(resp)
+        Ok(token)
     }
 
-    async fn authenticate(&mut self, api_key: &str) -> std::result::Result<(), GenericError> {
-        if api_key.split('|').collect::<String>().len() < 2 {
-            return Err("Invalid API key format".into());
-        }
-        self.handshake("", api_key).await?;
-        Ok(())
+    async fn authenticate(&self) -> std::result::Result<Option<String>, GenericError> {
+        let (username, password) = match &self.api_key {
+            Some(api_key) => {
+                if api_key.as_ref().split('|').collect::<String>().len() < 2 {
+                    return Err("Invalid API key format".into());
+                }
+                ("", api_key.as_ref())
+            }
```

Comment:
This was a remnant of the API key format we use in spice cloud, but we shouldn't restrict what API key format people use with the runtime itself.